src/components/chatcanvas/ChatCanvas.jsx

```diff
@@ -98,7 +99,7 @@ const ChatCanvas = ({
                   className="chat-button border border-input rounded-md flex p-3 mb-2 flex-col relative ms-5 dark:bg-black text-primary"
                   onClick={() => {
                     individualChatClickHandler();
-                    setSelectedFriend(friend);
+                    setSelectedFriend(friend.username);
```

Comment:
`selectedFriend` should be set to `friend` instead of `friend.username` so you can access the username, email, and profileURL properties from the `selectedFriend` object instead of only having access to the username.

Reply:
In lines 128 and 131 of `ChatBox.jsx`, the object properties are needed to properly display the username and user avatar.

Reply:
> `selectedFriend` should be set to `friend` instead of `friend.username` so you can access the username, email, and profileURL properties from the `selectedFriend` object instead of only having access to the username.

Code fixed. Thanks!